### ISSUE TITLE
Liquidsoap panics after some time with "Queue non-blocking queue #1 crashed with exception Bad file descriptor in select()"

### ISSUE THREAD

**Issue by MaPePeR** [savonet/liquidsoap#3702]
**Describe the bug**
After some time my liquidsoap crashed with this error (weirdly, that error isn't written to the logfile):

```
Queue non-blocking queue #1 crashed with exception Bad file descriptor in select()
Raised by primitive operation at Duppy.process.f in file "src/duppy.ml", line 211, characters 22-48
Re-raised at Duppy.process.f in file "src/duppy.ml", line 230, characters 12-19
Called from Duppy.process in file "src/duppy.ml", line 232, characters 4-8
Called from Duppy.queue.run in file "src/duppy.ml", line 337, characters 6-19
Called from Duppy.queue in file "src/duppy.ml", line 368, characters 7-11
Re-raised at Duppy.queue in file "src/duppy.ml", line 373, characters 8-44
Called from Tutils.create.(fun).process in file "src/core/tools/tutils.ml", line 186, characters 12-15
PANIC: Liquidsoap has crashed, exiting.,
Please report at: https://github.com/savonet/liquidsoapQueue non-blocking queue #1 crashed with exception Bad file descriptor in select()
Raised by primitive operation at Duppy.process.f in file "src/duppy.ml", line 211, characters 22-48
Re-raised at Duppy.process.f in file "src/duppy.ml", line 230, characters 12-19
Called from Duppy.process in file "src/duppy.ml", line 232, characters 4-8
Called from Duppy.queue.run in file "src/duppy.ml", line 337, characters 6-19
Called from Duppy.queue in file "src/duppy.ml", line 368, characters 7-11
Re-raised at Duppy.queue in file "src/duppy.ml", line 373, characters 8-44
Called from Tutils.create.(fun).process in file "src/core/tools/tutils.ml", line 186, characters 12-15
```

**To Reproduce**
Sadly I can't provide a reproducer, as my script is quite elaborate and it takes a while for the error to show. 
This is on windows. I switched to the pre-release because of #3685. It looks like this panic happens is in a similar place and I hope it is as easy to fix. If not I can try to provide more details (though I'm not very confident on that ability).

I feel like the issue only happened after I added a `rotate()` to intersperse jingles to my script, but I can't really confirm that.

**Expected behavior**
Not crash.

**Version details**

- OS: Windows
- Version: `liquidsoap-b573ac2-2.2.4-win64`

**Install method**
Downloaded asset from https://github.com/savonet/liquidsoap/releases/tag/rolling-release-v2.2.x

**Common issues**
The same panic was reported 8 years ago in #96, but I doubt it is related.

**Comment by toots** [comment]
Thanks for the report. Do you use external processes in your script?

**Comment by MaPePeR** [comment]
Yes. I call a python script from a custom protocol.

**Comment by MaPePeR** [comment]
I feel like there is something wrong with how liquidsoap waits for the end of a process. I also had to add `flush=True` to my `print` in Python to make my script more reliable, because sometimes the stdout in liquidsoap was empty even though the script was successful. It feels like liquidsoap does not truly wait for the process to finish and flush its output, which results in some weird race condition? Though I could not really reproduce it with a `time.sleep`. So its not really a "race" condition.

**Comment by toots** [comment]
Yes, your observation is correct. This was also the issue with the previous error. Something seems amiss where the process' stdout file descriptor is closed right as the process is terminated.

I'm working on switching our internal polling implementation away from `select` on windows so at least the whole process does not crash when this happens. I should have that ready soon.

**Comment by toots** [comment]
Alas it looks like the switch is more tricky than I thought. I'm gonna keep it in mind but it'll have to go through the `main` branch. I'll update here when it's available.
